### PR TITLE
Add event-based hints

### PR DESCRIPTION
**Pull request by aartaka**
This adds hints for HTML elements based on most frequent JS events -- `onclick` and `onfocus`.

How does it look to you?

### CODE REVIEW

File: source/element-hint.lisp [hunk 1]
```diff
@@ -239,20 +251,27 @@ identifier for every hinted element."
          :documentation "The body of the anchor tag."))
   (:accessor-name-transformer #'class*:name-identity))
 
-(defclass button-hint (hint) ())
+(define-class clickable-hint (hint) ())
```
Comment (jmercouris) [inline source/element-hint.lisp:254]: Why use define-class?

**Comment by jmercouris** [comment]
It looks great, thank you :-). Only my question about define-class vs defclass, otherwise OK to merge!

File: source/element-hint.lisp (hunk 1 above)
Comment (aartaka) [reply source/element-hint.lisp:254]: It's purely for consistency reasons, as I've said in commit message. Having `define-class` in all the hint definitions is more consistent and creates less cognitive load than having two different macros used for the same thing.

Comment (jmercouris) [reply source/element-hint.lisp:254]: Typically people attempt to use the most specific construct for their needs. For example one could use `(if xyz xy)` for everything, but instead it is preferred to use `(when xyz xy)`. In this case, we might as well use our own macro I guess.

**Pull request merged**